✨Feat: 회원가입 기능 구현완료

seoungjuyu opened the pull request:
### PR 타입(하나 이상의 PR 타입을 선택해주세요)
- [ ] 디자인 변경
- [x] 기능 추가
- [ ] 기능 삭제
- [ ] 버그 수정
- [ ] 의존성, 환경 변수, 빌드 관련 코드 업데이트

### 반영 브랜치
user -> main

### 변경 사항
- user url 연결
- signup 페이지에서 회원가입을 하면 admin 유저정보 저장
### 테스트 결과
1. signup 페이지
![화면 캡처 2022-10-19 113605](https://user-images.githubusercontent.com/110443976/196622866-62a48d29-173a-48e9-b4ec-8c11d26a8b96.png)
2.  회원가입을 하면 HttpResponse 으로 화면 변경(추후 로그인 구현할때 로그인 html로 나오게 할 예정)


![화면 캡처 2022-10-19 113547](https://user-images.githubusercontent.com/110443976/196623321-ae0791c2-da80-4204-b382-c543d514463e.png)

3.  admin user정보 저장
![화면 캡처 2022-10-19 113527](https://user-images.githubusercontent.com/110443976/196623453-71117d72-c7f6-4f2a-bc8a-5050dd1ca322.png)

seoungjuyu committed:
#1 회원가입 기능 구현

LeeHyunji commented:
제목을 커밋 컨벤션과 비슷하게 작업해주세요!

LeeHyunji commented on templates/user/signup.html:
```diff
@@ -0,0 +1,37 @@
+{% block content %}
+    <div class="container">
+        <div class="wrap">
+            <form class="form-area" method="POST" action="/signup/">
```
API 설계도 확인하면 user/sign-up URL에 API요청을 하는 것 같은데 기존 설계랑 다르네요? 확인부탁드립니다.

seoungjuyu replied:
다시 확인하겠습니다!

seoungjuyu commented:
넵!

LeeHyunji commented on user/urls.py:
```diff
@@ -0,0 +1,9 @@
+app_name = 'user'
+
+urlpatterns = [
+    path('signup/', views.signup, name='signup'),
```
여기도 기존 설계랑 다릅니다. 확인부탁드립니다.

seoungjuyu replied:
다시 확인하겠습니다!

LeeHyunji commented:
혹시 반영브런치가... 혹시 main에서  -> user인가요? 
merge 요청한 내용이랑 PR 내용이랑 다른데 내용도 확인부탁드립니다!

seoungjuyu commented:
넵!!

seoungjuyu committed:
feat #1 :hammer: API설계도 다른부분 수정 반영브런치 수정

API에 설계되어있는 url -> user/sign_up/ 으로 수정 sign-up으로 해야되나
오류가생겨 sign_up 으로 변경.
반영브런치 merge와 pr 부분이 달라서 수정.

close #1

LeeHyunji reviewed (commented):
sign-up도 될것 같은데 한번 다시 확인해보세요!

jetaime95 commented on user/models.py:
```diff
@@ -1,3 +1,10 @@
 # Create your models here.
+
+class User(AbstractUser): 
+    user = models.CharField(max_length=150, null=True)
+    name =models.CharField(max_length=20, null=True)
```
ERD 확인상 username, name, password 세가지를 사용하기로 했는데
user 모델을 추가로 만든 부분 확인 부탁 드립니다 !

seoungjuyu replied:
확인해서 수정하겠습니다!

son950610 commented:
위에서 다 말씀하셔서 나머진 좋은 거 같습니다.

seoungjuyu committed:
:hammer: user model과 db.sqlite  확인후 수정/ sign_up -> sign-up으로 수정 #1

- user model에 작성이 되어있던 user와name 에서 db.sqlit를 확인을 다시
해보니 django에서 기본으로 제공해주는 username이 있어서 model에 name만
수정
- API설계에 sign-up으로 되어있는데 views부분만 다르게 작성하면 되는
  부분이었는데 그부분을 정확히 확인을못하고 전부다 _ 로 실수함
  views부분빼고 - 로 수정

close #1

son950610 reviewed (commented):
확인했습니다~

LeeHyunji reviewed (commented):
승주님 git commit 메세지 컨벤션 지켜주시면 감사할 것 같습니다.
1. 제목은 <깃모지>랑 <태그명>명 두개다 작성해 주시면 좋을 것 같습니다! 
예시)
✨Feat : 회원가입 기능 구현
:bug:Bug : 회원가입 기능 중 아이디필드 중복해결 (username과 user)

2. PR 제목도 git commit 메세지 컨벤션을 따라 주시면 좋을 것 같습니다.  

3. 이슈번호로 `close`  사용할때 이슈에서 요청한 기능이 다 처리 되었는지를 항상 확인해야합니다.
지금 아래와 같이 #1 issue에 모든 기능이 완료되지 않았습니다. 
![Screen Shot 2022-10-19 at 8 23 35 PM](https://user-images.githubusercontent.com/12287842/196677649-de3d5270-a753-4c17-8b16-4fdcc713885c.png)
이럴경우 close를 해주시면 안됩니다. 마지막 기능까지 다 완료하신 후에 close 해주시면됩니다.

---
참고 git 커밋메세지 컨벤션
```
# 제목은 50자 이내 / 변경사항이 "무엇"인지 명확히 작성 / 끝에 마침표 금지
<깃모지><타입명>: <commit 제목> <#이슈번호>

#본문: 제목 다음에 한줄 공백 반드시 띄우기
본문(구체적인 내용)을 작성.
여러줄의 메세지를 작성할 땐 "-"로 구분 (한 줄은 72자 이내)

#꼬릿말(이슈 종료 등)
Close #이슈번호
```

LeeHyunji commented:
> 넵!! 더블체크하면서 컨벤션 지키겠습니다!!

사실 별거 아니라고 생각하실 수 있는데 생각보다 실무에서는 개발실력만큼이나 문서화를 잘 하는 것이 중요하다고 합니다!
Git을 여러 방법으로 **잘** **제대로** 사용하는 것이 협업할때도 마찬가지고, 나중에 취업할때 좋은 포트폴리오가 될 수 있기때문에 계속 반복해서 말씀드리는 건데.. 양해 부탁드립니다...! 죄송해요ㅠㅠ

seoungjuyu commented:
아님니다! 계속피드백 해주셔서 고쳐가면서 나아가지는것 같습니다! 저한테 하시는 피드백 달게 받겠습니다 ;)

jetaime95 reviewed (approved):
좋습니다 !

son950610 commented:
확인했습니다~

son950610 reviewed (approved):
확인했습니다~
